src/client/modules/my/dataUpload/dataUpload.ts

```diff
@@ -0,0 +1,127 @@
+import { LightningElement } from 'lwc';
+
+export default class DataUploader extends LightningElement {
+
+    /**
+     * Retrieves the data from the input fields and makes a query to upload it to the database api.
+     */
+    gatherDataAndSend() {
+
+        let playerData = []; //data for each player input
+
+        //get entered player data
+        for(let x = 0; x < 6; x++) {
+            playerData.push({
```

Comment:
This includes empty/null data, e.g:
```
matt 0
bob 1
"" null
"" null
"" null
"" null
```

IMO, we should have validateInput enforce that those are not valid values, and remove them from the back of the list until we get valid values.

e.g. the following would still be passed along, but the above example would not:
```
matt 0
"" null
bob 1
// notably: no more "" null entries at the end
```
then validateInput (correctly) rejects that input

Reply:
Processing and validation rules like this would also be great to include in a Jest test (once the validate method becomes pure)